update_language_files.py

```diff
@@ -156,13 +168,80 @@ def update_from_weblate():
     shutil.rmtree(tmp_dir, ignore_errors=True)
 
 
-def create_language_names_dict_in_file(language_codes: list) -> str:
-    """Create a py-file containing a dict of language names in different
-    flavours. The dict is used in the LanguageDialog to display the name of
+def create_completeness_dict():
+    """Create a simple dictionary indexed by language code and value that
+    indicate the completeness of the translation in percent.
+    """
+
+    print('Calculate completeness for each language in percent...')
+
+    result = {}
+
+    # each po file in the repository
+    for po_path in LOCAL_DIR.rglob('**/*.po'):
+        pof = polib.pofile(po_path)
+
+        result[po_path.stem] = pof.percent_translated()
+
+        pof.save()
+
+    # info
+    print(json.dumps(result, indent=4))
+
+    return result
+
+
+def create_languages_file():
+    """Create the languages.py file containing language names and the
+    completeness of their translation.
+
+    See the following functions for further details.
+    - ``update_language_names()``
+    - ``create_completeness_dict()``
+    """
+
+    # Convert language names dict to python code as a string
+    names = update_language_names()
+    stream = io.StringIO()
+    pprint.pprint(names, indent=2, stream=stream, sort_dicts=True)
+    stream.seek(0)
+    names = stream.read()
+
+    # the same with completeness dict
+    completeness = create_completeness_dict()
+    stream = io.StringIO()
+    pprint.pprint(completeness, indent=2, stream=stream, sort_dicts=True)
+    stream.seek(0)
+    completeness = stream.read()
+
+    with LANGUAGE_NAMES_PY.open('w', encoding='utf8') as handle:
+
+        handle.write('# Generated at {} with help of package "babel" '
+                     'and "polib".\n'.format(
+                         datetime.datetime.now().strftime('%c') ))
+        handle.write('# https://babel.pocoo.org\n')
+        handle.write('# https://github.com/python-babel/babel\n')
+
+        handle.write('\nnames = {\n')
+        handle.write(names[1:])
+
+        handle.write('\n')
+
+        handle.write('\ncompleteness = {\n')
+        handle.write(completeness[1:])
+
+    print(f'Result written to {LANGUAGE_NAMES_PY}.')
+
+
+def create_language_names_dict(language_codes: list) -> dict:
+    """Create dict of language names in different flavours.
+    The dict is used in the LanguageDialog to display the name of
     each language in the UI's current language and the language's own native
     representation.
     """
 
+    # We keep this import local because it is a rare case that this function
+    # will be called. This happens only if a new language is added to BIT.
     try:
         import babel
```

Comment:
I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install